app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ActiveRecord::Base
+  include EmailNormalization
+
+  belongs_to :manager, class_name: :User
+  has_many :employees, class_name: :User, foreign_key: :manager_id
+
+  validates :email, presence: true, format: /\A.+@.+\z/, uniqueness: true
```

Comment:
That's [a Ruby peculiarity](http://ruby-doc.org/core-2.2.0/doc/regexp_rdoc.html):

> * \Z - Matches end of string. If string ends with a newline, it matches just before newline
> * \z - Matches end of string